.buildkite/generate_pipeline.py

```diff
@@ -0,0 +1,96 @@
+import yaml
+import json
+import pathlib
+import sys
+import os
+
+CONTAINER_VERSION = "v12"
+DOCKER_PLUGIN_VERSION = "v3.8.0"
+
+X86_AGENT_TAGS = os.getenv('X86_LINUX_AGENT_TAGS')
+AARCH64_AGENT_TAGS = os.getenv('AARCH64_LINUX_AGENT_TAGS')
+DOCKER_PLUGIN_CONFIG = os.getenv('DOCKER_PLUGIN_CONFIG')
+
+PLATFORMS = ['x86_64.metal', 'arm.metal']
+OSS = ['linux']
+
+ROOT_PATH = pathlib.Path(__file__).parent.resolve()
+
+
+def check_step(step):
+    assert step.get('label'),\
```

Comment:
We can define a list of required keys to avoid the duplicate code. Also we can use a dictionary for the keys that accept only some particular values.